http/src/request/guild/create_guild.rs

```diff
@@ -67,6 +71,104 @@ struct CreateGuildFields {
     verification_level: Option<VerificationLevel>,
 }
 
+/// Fillable struct to send new roles to a guild create.
+#[derive(Clone, Debug, Default)]
+pub struct RoleFragment {
+    pub color: Option<u32>,
+    pub hoist: bool,
+    pub id: Option<RoleId>,
+    pub mentionable: bool,
+    pub name: Option<String>,
+    pub permissions: Option<Permissions>,
+    pub position: Option<i64>,
+}
+
+#[derive(Clone, Debug)]
+pub enum GuildChannelFragment {
+    Category(CategoryChannelFragment),
+    Text(TextChannelFragment),
+    Voice(VoiceChannelFragment),
+}
+
+/// Fillable struct to send new category channels to a guild create.
+#[derive(Clone, Debug)]
+pub struct CategoryChannelFragment {
+    pub id: ChannelId,
+    pub name: String,
+    pub permission_overwrites: Option<Vec<PermissionOverwrite>>,
+}
+
+impl Into<CategoryChannel> for CategoryChannelFragment {
+    fn into(self) -> CategoryChannel {
+        CategoryChannel {
+            guild_id: Some(GuildId(1)),
+            id: self.id,
+            kind: ChannelType::GuildCategory,
+            name: self.name,
+            nsfw: false,
+            parent_id: None,
+            permission_overwrites: self.permission_overwrites.unwrap_or_else(Vec::new),
+            position: 0,
+        }
+    }
+}
+
+/// Fillable struct to send new text channels to a guild create.
+#[derive(Clone, Debug)]
+pub struct TextChannelFragment {
+    pub name: String,
+    pub nsfw: bool,
+    pub parent_id: Option<ChannelId>,
+    pub permission_overwrites: Option<Vec<PermissionOverwrite>>,
+    pub rate_limit_per_user: Option<u64>,
+    pub topic: Option<String>,
+}
+
+impl Into<TextChannel> for TextChannelFragment {
+    fn into(self) -> TextChannel {
+        TextChannel {
+            guild_id: Some(GuildId(1)),
+            id: ChannelId(501),
```

Comment:
Does 501 have a specific meaning here? If so there should be a comment here for anyone else that comes across it